### ISSUE TITLE
Sample Code for non-PFX certificate usage

### ISSUE THREAD

**Issue by jkoplo** [aws-samples/iot-dotnet-publisher-consumer#12]
I have sample code for connecting to AWS IoT using the cert files directly (and not using OpenSSL to convert to a PFX). I'd like to contribute this for connecting both with M2MQTT and MQTTNet.

It resolves some open issues in other projects ([#927](https://github.com/chkr1011/MQTTnet/issues/927), [#721](https://github.com/chkr1011/MQTTnet/issues/721)) but since it's AWS IoT specific I think it probably belongs in this repo.

My only hesitation is that the README is already pretty long and adding more sections/branches for different ways of handling certs and also implementations with MQTTNet alongside M2MQTT is a pretty big restructure. There's also some folder restructuring that might be needed.

Basically there would be 4 factors, leading to 16 different examples if I follow the standard of one csproj/folder per example:

1. Core OR Framework
2. Publisher OR Subscriber
2. Certs OR pfx
3. MQTTNet OR M2MQTT

Would you be okay with me reorganizing the folder structure and maybe combining examples into a few projects? I think Publisher and Subscriber could easily be in the same project and the Cert/pfx code could just be commented out or branched based on a boolean.

Thoughts?
